codex/sales/states/downloading.nim

```diff
@@ -31,6 +31,11 @@ method onSlotFilled*(state: SaleDownloading, requestId: RequestId,
                      slotIndex: UInt256): ?State =
   return some State(SaleFilled())
 
+# override onError so that the slot is reprocess in the case of an unhandled
+# exception
+method onError*(state: SaleDownloading, error: ref CatchableError): ?State =
+  return some State(SaleErrored(error: error, reprocessSlot: true))
```

Comment:
I know we discussed this before, but I can't remember the conclusion. Did we want to try a slot again if it failed to download the first time? I would think that we don't want to do this, because there's no reason to assume that it would work a second time around, so it's better to focus on other slots?

Reply:
Yes, that's correct. We did discuss it and I have it in my notes that it should
be updated so that errors are not reprocessed. But it was so long ago, I forgot
to do it. Great catch, thank you!